src/masternodes/rpc_tokens.cpp

```diff
@@ -639,45 +639,77 @@ UniValue getcustomtx(const JSONRPCRequest& request)
+                        "",
                         {
-                            {"", RPCArg::Type::OBJ, RPCArg::Optional::OMITTED, "",
-                                {
-                                    {"txid", RPCArg::Type::STR_HEX, RPCArg::Optional::NO, "The transaction id"},
-                                    {"vout", RPCArg::Type::NUM, RPCArg::Optional::NO, "The output number"},
-                                },
-                            },
+                            {"txid", RPCArg::Type::STR_HEX, RPCArg::Optional::NO, "The transaction id"},
+                            {"vout", RPCArg::Type::NUM, RPCArg::Optional::NO, "The output number"},
                         },
                     },
-               },
-               RPCResult{
-                       "\"hash\"                  (string) The hex-encoded hash of broadcasted transaction\n"
-               },
-               RPCExamples{
-                       HelpExampleCli("minttokens", "10@symbol")
-                       + HelpExampleCli("minttokens",
-                                      "10@symbol '[{\"txid\":\"id\",\"vout\":0}]'")
-                       + HelpExampleRpc("minttokens", "10@symbol '[{\"txid\":\"id\",\"vout\":0}]'")
-               },
-    }.Check(request);
+                },
+            }, {"to", RPCArg::Type::STR, RPCArg::Optional::OMITTED, "Address to mint tokens to"},
+                          },
+        RPCResult{"\"hash\"                  (string) The hex-encoded hash of broadcasted transaction\n"},
+        RPCExamples{HelpExampleCli("minttokens", "10@symbol") +
+                    HelpExampleCli("minttokens", R"(10@symbol '[{"txid":"id","vout":0}]')") +
+                    HelpExampleCli("minttokens", R"(10@symbol '[{"txid":"id","vout":0}] address')") +
+                    HelpExampleRpc("minttokens", R"(10@symbol '[{"txid":"id","vout":0}]')")},
+    }
+        .Check(request);
 
     if (pwallet->chain().isInitialBlockDownload()) {
         throw JSONRPCError(RPC_CLIENT_IN_INITIAL_DOWNLOAD,
                            "Cannot mint tokens while still in Initial Block Download");
     }
     pwallet->BlockUntilSyncedToCurrentChain();
 
-    const CBalances minted = DecodeAmounts(pwallet->chain(), request.params[0], "");
-    UniValue const & txInputs = request.params[1];
+    CBalances minted;
+    UniValue txInputs;
+    CScript to;
+
+    if (request.params[0].isObject()) {
+        auto optionsObj = request.params[0].get_obj();
+        minted = DecodeAmounts(pwallet->chain(), optionsObj["amounts"].get_array(), "");
+
+        if (optionsObj.exists("inputs"))
+            txInputs = optionsObj["inputs"].get_array();
+
+        if (optionsObj.exists("to"))
+            to = DecodeScript(optionsObj["to"].get_str());
+    }
+    else {
+        minted   = DecodeAmounts(pwallet->chain(), request.params[0], "");
+        txInputs = request.params[1];
+
+        if (request.params.size() > 2)
```

Comment:
I wouldn't put `to` as last param as it means you need to supply `inputs` before `to`. I would avoid adding `to` if it is not a JSON object.

Also, we need to prevent having to as empty address supplied so that it doesn't accidentally go to empty address!

Reply:
I think to needs to be after inputs to not break anything, interface needs to remain the same for services using this. Empty inputs can be provided.

Reply:
Ahhh, yes. I forgot for the moment ...